### PR TITLE
Add ability to use Dev Containers locally

### PR DESCRIPTION
**Pull request by nikhitasharma**
### Summary
Adds ability to run npm commands locally on a Docker container using VScode devcontainers setup

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -175,6 +176,33 @@ npm install
...
 
+<h2 id="docker-local-development">
+  Local Development using Docker (Dev Containers)
+</h2>
+
+1. Install docker https://www.docker.com/.
+2. Install [Microsoft VS Code](https://code.visualstudio.com/Download)
+3. Clone this repo.
+4. Open this repo in VS Code.
+5. Install the Dev Container extension.
```
Comment (sdadn) [inline README.md:187]: We can add the extension ID so consumers can install the exact extension you're talking about. 
```suggestion
5. Install the Dev Container extension. In the `Extensions` tab, you can search: `ms-vscode-remote.remote-containers`.
```

File: README.md [hunk 2]
```diff
@@ -175,6 +176,32 @@ npm install
...
+    - Navigate to View -> Extension  -> Search for and install _Dev Containers_
```
[the remark this answers is not in the document]
Comment (nikhitasharma) [reply README.md]: added

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1 @@
+FROM docker.cernerrepos.net/ion/ion-node:5
```
[the remark this answers is not in the document]
Comment (nikhitasharma) [reply .devcontainer/Dockerfile]: fixed

File: README.md [hunk 4]
```diff
@@ -175,6 +176,33 @@ npm install
...
+    - More information on [Dev Containers](https://code.visualstudio.com/docs/devcontainers/containers)
+6. Build the dev container.
+    - Make sure you are connected to the Cerner VPN when building this container the first time.
+    - Navigate to View -> Command Palette -> Open Workspace In Container or Reopen In Container.
+7. You're now running in a dev container.  Use the terminal of the dev container to issue any npm commands.
+8. Before running any WDIO tests (`npm run test` or `npm run wdio`), make sure to perform the following steps:
+    - Open a new terminal (outside the dev container) and navigate to  ".devcontainer/" path in your repository.
+    - Execute the command `"docker compose -f docker-compose-wdio.yml up"`. Selenium hub should spin up. Leave this running in the background. If you see errors saying "container name already exists", run `"docker container prune"` command followed by selecting "y" to clear up any unused containers and try running the previous command again.
+    - Edit the root package.json file WDIO scripts to include `--disableSeleniumService=true` flag. This will disable the selenium service from spinning up again.
```
Comment (sdadn) [inline README.md:197]: Instead of this, maybe we can add a `docker:wdio` script that uses this flag?

File: README.md [hunk 5]
```diff
@@ -175,6 +176,33 @@ npm install
...
+5. Install the Dev Container extension.
+    - Navigate to View -> Extension  -> Search for and install _Dev Containers_ (or "ms-vscode-remote.remote-containers")
+    - More information on [Dev Containers](https://code.visualstudio.com/docs/devcontainers/containers)
+6. Build the dev container.
+    - Make sure you are connected to the Cerner VPN when building this container the first time.
+    - Navigate to View -> Command Palette -> Open Workspace In Container or Reopen In Container.
+7. You're now running in a dev container.  Use the terminal of the dev container to issue any npm commands.
+8. Before running any WDIO tests (`npm run test` or `npm run wdio`), make sure to perform the following steps:
+    - Open a new terminal (outside the dev container) and navigate to  ".devcontainer/" path in your repository.
+    - Execute the command `"docker compose -f docker-compose-wdio.yml up"`. Selenium hub should spin up. Leave this running in the background. If you see errors saying "container name already exists", run `"docker container prune"` command followed by selecting "y" to clear up any unused containers and try running the previous command again.
```
Comment (sdadn) [inline README.md:196]:
```suggestion
    - Execute the command `docker compose -f docker-compose-wdio.yml up`. Selenium hub should spin up. Leave this running in the background. If you see errors saying "container name already exists", run `docker container prune` command followed by pressing `y` to clear up any unused containers and try running the previous command again.
```

**Review by lqdev00 — APPROVED** [review]
Looks good to me.

File: README.md (hunk 4 above)
Comment (nikhitasharma) [reply README.md:197]: I will add in a new scripts `wdio:docker` and `test:docker` which would use the disableSeleniumService flag set to true.  I would like to still keep this explanation in the readme so they understand the process.

**Review by seanturk — APPROVED** [review]

**Review by Veera81 — APPROVED** [review]

File: .devcontainer/devcontainer.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+    "name": "TerraCore",
```
[the remark this answers is not in the document]
Comment (nikhitasharma) [reply .devcontainer/devcontainer.json:2]: This is just a container name to display on the vscode UI, so it can be anything which makes sense to the devs.

**Review by sdadn — APPROVED** [review]

**Review by seanturk — APPROVED** [review]

**Pull request merged**